### PR TITLE
Nodejs week2

### PR DESCRIPTION
**Pull request by melo4yxa77796**

### CODE REVIEW

File: api/src/index.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
+
 import "dotenv/config";
 import express from "express";
 import cors from "cors";
-import bodyParser from "body-parser";
-import knex from "./database_client.js";
+import bodyParser from "body-parser";
```
Comment (urbanogilson) [inline api/src/index.js]: Please use Prettier

```suggestion
import bodyParser from "body-parser";
```

File: api/src/routers/meals.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+const mealsRouter = express.Router();
+
+
+  mealsRouter.get("/", async (req, res) => {
+    try {
+      const meals = await knex("Meal").select("*"); // Select all columns from the Meal table
+      res.json(meals);
+    } catch (error) {
+      console.error("Error retrieving meals:", error);
+      res.status(500).json({ error: "Error retrieving meals from the database." });
```
Comment (urbanogilson) [inline api/src/routers/meals.js]: It would be nice to try to move this logic to a middleware

File: api/src/routers/meals.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
+import express from 'express';
+import knex from '../database_client.js'; 
+
+const mealsRouter = express.Router();
+
+
+  mealsRouter.get("/", async (req, res) => {
+    try {
+      const meals = await knex("Meal").select("*"); // Select all columns from the Meal table
```
Comment (urbanogilson) [inline api/src/routers/meals.js]: One suggestion is to not make implicit methods or values ​​explicit.

The code in this case is clear, so no comments are required.

```suggestion
      const meals = await knex("Meal");
```

File: api/src/routers/meals.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+      console.error("Error retrieving meals:", error);
+      res.status(500).json({ error: "Error retrieving meals from the database." });
+    }
+  });
+  
+  
+
+  mealsRouter.get("/:id", async (req, res) => {
+    const { id } = req.params;
+    try {
```
Comment (urbanogilson) [inline api/src/routers/meals.js]:
```suggestion
    try {
    const { id } = req.params;
```

File: api/src/routers/meals.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      const meal = await knex("Meal").where({ id }).first(); // Select meal by ID
```
Comment (urbanogilson) [inline api/src/routers/meals.js]: You named your table `Meal`, checkout this page for SQL table/column name conventions: [https://brainstation.io/learn/sql/naming-conventions](https://brainstation.io/learn/sql/naming-conventions)


```suggestion
      const meal = await knex("meal").where({ id }).first();
```

File: api/src/routers/meals.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    const newMeal = await knex("Meal").insert({
+      title,
+      description,
+      location,
+      when: formattedDate, // Store the normalized date
+      max_reservations,
+      price,
+      created_date: new Date().toISOString().slice(0, 10) // Set current date
```
Comment (urbanogilson) [inline api/src/routers/meals.js]: The Date time field requires a special representation in SQL. You can explore knex helpers like https://knexjs.org/guide/utility.html#now to solve these common issues

```suggestion
      created_date: knex.fn.now()
```

File: api/src/routers/reservations.js [hunk 7]
```diff
@@ -0,0 +1,113 @@
+
+import express from 'express';
+import knex from '../database_client.js'; // Adjust the path if necessary
+
+const reservationsRouter = express.Router();
+
+// Retrieve all reservations
+reservationsRouter.get("/", async (req, res) => {
+  try {
+    const reservations = await knex("Reservation").select('*'); // Retrieve all reservations
```
Comment (urbanogilson) [inline api/src/routers/reservations.js]:
```suggestion
    const reservations = await knex("Reservation");
```

**Review by urbanogilson — COMMENTED** [review]
Good job! Left inline comments

File: api/src/index.js (hunk 1 above)
Comment (melo4yxa77796) [reply api/src/index.js]: 👍

File: api/src/routers/meals.js (hunk 2 above)
Comment (melo4yxa77796) [reply api/src/routers/meals.js]: I did it

File: api/src/routers/meals.js (hunk 3 above)
Comment (melo4yxa77796) [reply api/src/routers/meals.js]: 👍

File: api/src/routers/meals.js (hunk 5 above)
Comment (melo4yxa77796) [reply api/src/routers/meals.js]: Did I understand correctly from the article that it is more correct to name it with a small letter?

File: api/src/routers/reservations.js (hunk 7 above)
Comment (melo4yxa77796) [reply api/src/routers/reservations.js]: 👍

File: api/src/routers/meals.js (hunk 4 above)
Comment (melo4yxa77796) [reply api/src/routers/meals.js]: I moved it

File: api/src/routers/meals.js (hunk 6 above)
Comment (melo4yxa77796) [reply api/src/routers/meals.js]: i did it

**Pull request merged**